contracts/truefi2/TrueCreditAgency.sol

```diff
@@ -361,23 +364,62 @@ contract TrueCreditAgency is UpgradeableClaimable {
 
     function poke(ITrueFiPool2 pool) public {
         uint256 bitMap = usedBucketsBitmap;
-        CreditScoreBucket[256] storage creditScoreBuckets = buckets[pool];
         uint256 timeNow = block.timestamp;
         uint256 partialRate = riskPremium.add(utilizationAdjustmentRate(pool));
 
+        for (uint16 i = 0; i <= MAX_CREDIT_SCORE; (i++, bitMap >>= 1)) {
+            if (bitMap & 1 == 0) {
+                continue;
+            }
+
+            _pokeSingleBucket(pool, uint8(i), timeNow, partialRate);
+        }
+    }
+
+    function pokeSingleBucket(ITrueFiPool2 pool, uint8 bucketNumber) internal {
+        uint256 timeNow = block.timestamp;
+        uint256 partialRate = riskPremium.add(utilizationAdjustmentRate(pool));
+
+        _pokeSingleBucket(pool, bucketNumber, timeNow, partialRate);
+    }
+
+    function _pokeSingleBucket(
+        ITrueFiPool2 pool,
+        uint8 bucketNumber,
+        uint256 timeNow,
+        uint256 partialRate
+    ) internal {
+        CreditScoreBucket storage bucket = buckets[pool][bucketNumber];
+
+        poolTotalInterest[pool] = poolTotalInterest[pool].add(
+            bucket.rate.mul(1e23).mul(bucket.totalBorrowed).mul(timeNow.sub(bucket.timestamp)).div(365 days)
+        );
+
+        bucket.cumulativeInterestPerShare = bucket.cumulativeInterestPerShare.add(
+            bucket.rate.mul(1e23).mul(timeNow.sub(bucket.timestamp)).div(365 days)
+        );
+        bucket.rate = _currentRate(partialRate, _creditScoreAdjustmentRate(bucketNumber));
+        bucket.timestamp = uint128(timeNow);
+    }
+
+    function poolCreditValue(ITrueFiPool2 pool) external override view returns (uint256) {
+        uint256 bitMap = usedBucketsBitmap;
+        CreditScoreBucket[256] storage creditScoreBuckets = buckets[pool];
+        uint256 timeNow = block.timestamp;
+        uint256 value = poolTotalInterest[pool].div(1e27);
+
         for (uint16 i = 0; i <= MAX_CREDIT_SCORE; (i++, bitMap >>= 1)) {
             if (bitMap & 1 == 0) {
                 continue;
             }
 
             CreditScoreBucket storage bucket = creditScoreBuckets[i];
 
-            bucket.cumulativeInterestPerShare = bucket.cumulativeInterestPerShare.add(
-                bucket.rate.mul(1e23).mul(timeNow.sub(bucket.timestamp)).div(365 days)
+            value = value.add(bucket.totalBorrowed).add(
+                bucket.rate.mul(1e23).mul(bucket.totalBorrowed).mul(timeNow.sub(bucket.timestamp)).div(365 days).div(1e27)
```

Comment:
I think we could pull out any multiplicative factors that stay constant in the bucket loop, to save some gas and improve readability.

Assuming we factor out bucket interest per share, this hopefully just leaves the overall division by 1e27